src/main/java/org/fundacionjala/virtualassistant/repository/WhisperClient.java

```diff
@@ -0,0 +1,110 @@
+package org.fundacionjala.virtualassistant.repository;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.ShortBuffer;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+import javax.sound.sampled.UnsupportedAudioFileException;
+
+import io.github.givimad.whisperjni.WhisperContext;
+import io.github.givimad.whisperjni.WhisperFullParams;
+import io.github.givimad.whisperjni.WhisperJNI;
+import io.github.givimad.whisperjni.WhisperSamplingStrategy;
+import lombok.Data;
+
+@Data
+public class WhisperClient implements ASRClient {
+
+    private static final Path MODEL_WHISPER = Path.of("ggml-tiny.bin");
+
+    private WhisperJNI whisper;
+
+    private String transcription;
+
+    public WhisperClient() throws IOException{
+        WhisperJNI.loadLibrary();
+        whisper = new WhisperJNI();
+    }
+
+    @Override
+    public void convertToText(Path audioFile) throws IOException, UnsupportedAudioFileException {
+        validateModelAndAudioFilesExist(audioFile);
+        float[] audioFrequency = regulateFrequency(audioFile);
+        List<String> transcriptions = transcribeSegments(audioFrequency);
+        transcription = String.join(" ", transcriptions);
+    }
+
+    private List<String> transcribeSegments(float[] samples) {
+        List<String> transcriptions = new ArrayList<>();
+
+        try (WhisperContext ctx = whisper.init(MODEL_WHISPER)) {
+            WhisperFullParams params = new WhisperFullParams(WhisperSamplingStrategy.BEAN_SEARCH);
```

Comment:
I will be better to use a unique instance of whisper, so please move this call to have a singleton instance of Whisper